### ISSUE TITLE
errors for failed reconstruction jobs

### ISSUE THREAD

**Issue by qiuyixmm** [Magdoll/Cogent#51]
@Magdoll 
    when Coding Genome Reconstruction jobs were runing,some jobs were failed. If the file in.fa in failed_job_job_directory was less than 100kb, i could add --nx_cycle_detection and tried bigger k to solve the problems.
     however, if the file in.fa in failed_job_job_directory was bigger than 100kb, --nx_cycle_detection and bigger k didnt work. In this case, some directories were produced in failed_job_job_directory whose name begin with "split_". And then a CYCLE_DETECTED file  whose size was 0 would be produced in one of "split_n" directories. Finally, there were no new files produced though  Coding Genome Reconstruction jobs were runing. An example was given below:
a. cd ~/12495_0/  &  ls 
       hello.log
       in.fa
       in.weights
       split_0
       split_1
       split_2
       split_3
       split_4
       split_5

b. cd split_0  &  ls
     CYCLE_DETECTED    #size is 0
      in.fa
      in.trimmed.fa
      in.weights

c. cd ../split_1  &  ls  
     in.fa
     in.weights    # split_2 - split_5 also only have in.fa and in.weights

**Comment by Magdoll** [comment]
Hi @qiuyixmm ,

This is a case where actually one of the splits (large inputs get split into sub-tasks and merged at the end) end up with a cycle. An empty outfile `CYCLE_DETECTED` is produced whenever cycles are detected and the job is aborted. I may be not dealing with failed jobs properly in split cases.

Would you be willing to share one example of a large (split-up) input that had a cycle in one of the splits?

If you provide an email, I can send a request to have you upload the file securely.

Thanks
--Liz

**Comment by qiuyixmm** [comment]
Hello @Magdoll  
 this is my email address :  xumingmin@mail.kiz.ac.cn
   there are about 1 to 2 failed job directories for one sample. I am going to remove these failed directoires if my described problem was not solved. do you think the final produed fake genome has a significant influcence to collapse isoforms without genome ?
     Thanks!

**Comment by Magdoll** [comment]
Hi @qiuyixmm ,

A request for file upload has been sent to the email.

I think it would be helpful to fix these failed jobs because these are the jobs that have a large number of input transcripts. Excluding them from your dataset means excluding results of (potentially) dominant transcripts.

Also note: I will be traveling for work the entire next week, I will do my best to solve your problem as soon as possible.

--Liz

**Comment by Magdoll** [comment]
Hi @qiuyixmm ,

I looked at the input sequences. It is extremely repetitive, which is why it originally failed. I lifted the limit of kmer size (`-k` parameter) which used to be capped at 200. This input set is so repetitive that I was not able to complete it at `-k 200`, but I tried `-k 300` and it was able to complete, however larger kmer sizes often means less reconstruction (hence, more sequences in the output file `cogent2.fa`).

I've committed a new version v3.7 to Cogent that allows you to use larger `-k`. 

The result has been emailed to you.

--Liz
